src/config.js

```diff
@@ -334,7 +335,7 @@ export default class Config {
 
     this.disablePrepublish = !!opts.disablePrepublish;
 
-    this.nonInteractive = !!opts.nonInteractive;
+    this.nonInteractive = opts.nonInteractive != null ? !!opts.nonInteractive : isCi || !process.stdout.isTTY;
```

Comment:
Wow, this is some hard-to-follow logic! How about this:
```
this.nonInteractive = !!opts.nonInteractive || isCi || !process.stdout.isTTY;
```

I know understand what you are trying to do (honor when `non-interative` is forced to `false`) but it is a case that cannot happen currently. What do you think?